src/sync.rs

```diff
@@ -705,9 +705,10 @@ impl<T: Copy + PartialEq> PartialEq for LockFreeFrozenVec<T> {
             return false;
         }
 
-        // since the lengths are the same, just check the elements in order
+        // Since the lengths are the same, just check the elements in order
+        // this assumes that the vectors don't change while equality is being checked
```

Comment:
This comment is incorrect: this assumption is not being made

Reply:
```suggestion
```